src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -528,9 +528,14 @@ int32 AuraEffect::CalculateAmount(Unit* caster)
             {
                 Item* item1 = caster->ToPlayer()->GetWeaponForAttack(BASE_ATTACK);
                 Item* item2 = caster->ToPlayer()->GetWeaponForAttack(OFF_ATTACK);
+                Item* shield = caster->ToPlayer()->GetShield();
```

Comment:
instead of this, can't you just check if `item2->GetTemplate()->InventoryType == INVTYPE_SHIELD`

Reply:
As the shield would always be in the item2 slot anyway

Reply:
`GetWeaponForAttack()` returns null for shields, so if equipping a shield then item2 doesn't exist. This is why I used `GetShield()`

Reply:
got it